src/util/GeoSparqlHelpers.h

```diff
@@ -32,48 +25,47 @@ static constexpr double invalidCoordinate =
 // TODO: Make the SPARQL expressions work for function pointers or
 // std::function.
 std::pair<double, double> parseWktPoint(const std::string_view point);
-double wktLongitudeImpl(const std::string_view point);
-double wktLatitudeImpl(const std::string_view point);
-double wktDistImpl(const std::string_view point1,
-                   const std::string_view point2);
+double wktLongitudeImpl(GeoPoint point);
+double wktLatitudeImpl(GeoPoint point);
 double wktDistImpl(GeoPoint point1, GeoPoint point2);
 
 }  // namespace detail
 
-// Parse the longitude coordinate from a WKT point (it's the first coordinate).
-inline auto wktLongitude = [](const std::optional<std::string>& point) {
-  if (!point.has_value()) {
-    return std::numeric_limits<double>::quiet_NaN();
+// Return the longitude coordinate from a WKT point.
+class WktLongitude {
+ public:
+  double operator()(const std::optional<GeoPoint>& point) const {
+    if (!point.has_value()) {
+      return std::numeric_limits<double>::quiet_NaN();
+    }
+    return detail::wktLongitudeImpl(point.value());
   }
-  return detail::wktLongitudeImpl(point.value());
 };
 
-// Parse the latitude coordinate from a WKT point (it's the second coordinate).
-inline auto wktLatitude = [](const std::optional<std::string>& point) {
-  if (!point.has_value()) {
-    return std::numeric_limits<double>::quiet_NaN();
+// Return the latitude coordinate from a WKT point.
+class WktLatitude {
+ public:
+  double operator()(const std::optional<GeoPoint>& point) const {
+    if (!point.has_value()) {
+      return std::numeric_limits<double>::quiet_NaN();
+    }
+    return detail::wktLatitudeImpl(point.value());
```

Comment:
No need for the `Impl`, write `point.value().getLat` direclty here, and clean the rest up.